### PR TITLE
fix spine wasm api array bind

### PR DESCRIPTION
**Pull request by Canvasfull**
Re: #

### Changelog

*fix spine wasm API array bind.

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/spine/lib/spine-define.ts [hunk 1]
```diff
@@ -42,6 +42,50 @@ function overrideDefineArrayProp (prototype: any, getPropVector: any, name: stri
...
 
+function overrideDefineArrayArrayProp (prototype: any, getPropVector: any, name: string): void {
+    Object.defineProperty(prototype, name, {
+        get (): any[] {
+            const array: any[] = [];
+            const vectors = getPropVector.call(this);
+            const count = vectors.size();
+            for (let i = 0; i < count; i++) {
+                const vectorI = vectors.get(i);
+                const countJ = vectorI.size()
```
Comment (SantyWang) [inline cocos/spine/lib/spine-define.ts]: Fix eslint error.

File: native/cocos/editor-support/spine/Vector.h [hunk 2]
```diff
@@ -158,6 +158,12 @@ class SP_API Vector : public SpineObject {
         return -1;
     }
 
+    inline const T &operator[](size_t inIndex) const {
+        assert(inIndex < _size);
```
Comment (SantyWang) [inline native/cocos/editor-support/spine/Vector.h]: Why add this function?

Comment (Canvasfull) [reply native/cocos/editor-support/spine/Vector.h]: Spine does not provide const Vector operator[] in C++ to go to TS API, which needs to be converted to std::vector and needs to be accessed. I added this API after communicating with Jianhua.
Screenshot 2023-08-22 at 16 26 58

Comment (Canvasfull) [reply native/cocos/editor-support/spine/Vector.h]: change copy code Vector<String> attachmentNames = obj.getAttachmentNames();

**Review by SantyWang — APPROVED** [review]

File: cocos/spine/lib/spine-define.ts [hunk 3]
```diff
@@ -42,6 +43,50 @@ function overrideDefineArrayProp (prototype: any, getPropVector: any, name: stri
...
+                const countJ = vectorI.size();
+                const arrayJ: any[] = [];
+                for (let j = 0; j < countJ; j++) {
+                    arrayJ.push(vectorI.get(j));
+                }
+                array.push(arrayJ);
+            }
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+            return array;
+        },
```
Comment (dumganhar) [inline cocos/spine/lib/spine-define.ts:63]: Do we need to add setter for this?

File: cocos/spine/lib/spine-define.ts [hunk 4]
```diff
@@ -42,6 +43,50 @@ function overrideDefineArrayProp (prototype: any, getPropVector: any, name: stri
...
+            for (let i = 0; i < count; i++) {
+                const objPtr = vectors.get(i);
+                array.push(objPtr);
+            }
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+            return array;
+        },
+        set (value: any[]) {
+            const vectors = new Type();//spine.wasmUtil.wasm.VectorFloat();
+            for (const obj of value) {
```
Comment (dumganhar) [inline cocos/spine/lib/spine-define.ts]: Is `vectors.reserve()` exported to JS ? If so, it's better to call it after `new Type()`.
```js
const vectors = new Types();
vectors.reserve(value.length);
```

File: cocos/spine/lib/spine-define.ts (hunk 3 above)
Comment (Canvasfull) [reply cocos/spine/lib/spine-define.ts:63]: Currently there is only one place to use and only get.
Screenshot 2023-08-23 at 10 39 34

File: cocos/spine/lib/spine-define.ts [hunk 5]
```diff
@@ -582,15 +604,14 @@ function overrideProperty_MeshAttachment (): void {
...
-            getter: prototype.getEdges,
-        },
     ];
     propertyPolyfills.forEach((prop): void => {
         js.getset(prop.proto, prop.property, prop.getter, prop.setter);
     });
+    overrideDefineArrayProp(prototype, prototype.getRegionUVs, 'regionUVs');
+    overrideDefineArrayProp(prototype, prototype.getUVs, 'uvs');
+    overrideDefineArrayProp(prototype, prototype.getTriangles, 'triangles');
+    overrideDefineArrayProp(prototype, prototype.getTriangles, 'edges');
```
Comment (dumganhar) [inline cocos/spine/lib/spine-define.ts]: Mistake here. It should be `prototype.getEdges` here.

**Review by dumganhar — APPROVED** [review]

**Pull request merged**